batch_test.go

```diff
@@ -0,0 +1,251 @@
+	group, err := cluster.DaemonAt(0).NewGroup("test-remove-keys", 3000000, getter)
+	require.NoError(t, err)
+	for i := 1; i < 3; i++ {
+		_, err := cluster.DaemonAt(i).NewGroup("test-remove-keys", 3000000, getter)
+		require.NoError(t, err)
+	}
+
+	keys := []string{"key1", "key2", "key3"}
+
+	// First, populate the cache by getting each key
+	for _, key := range keys {
+		var value string
+		err := group.Get(ctx, key, transport.StringSink(&value))
+		require.NoError(t, err)
+		assert.Equal(t, fmt.Sprintf("value-%s", key), value)
+	}
+
+	// Verify getter was called for each key
+	for _, key := range keys {
+		assert.Equal(t, 1, callCount[key], "getter should be called once for %s", key)
+	}
+
+	// Now remove all keys using variadic signature
+	err = group.RemoveKeys(ctx, "key1", "key2", "key3")
+	require.NoError(t, err)
+
+	// Fetch again - getter should be called again since keys were removed
+	for _, key := range keys {
+		var value string
+		err := group.Get(ctx, key, transport.StringSink(&value))
+		require.NoError(t, err)
+	}
+
+	// Verify getter was called again for each key
+	for _, key := range keys {
+		assert.Equal(t, 2, callCount[key], "getter should be called twice for %s after removal", key)
+	}
+}
+
+func TestRemoveKeysEmpty(t *testing.T) {
+	ctx := context.Background()
+
+	err := cluster.Start(ctx, 2, groupcache.Options{})
+	require.NoError(t, err)
+	defer func() { _ = cluster.Shutdown(ctx) }()
+
+	getter := groupcache.GetterFunc(func(ctx context.Context, key string, dest transport.Sink) error {
+		return dest.SetString("value", time.Now().Add(time.Minute))
+	})
+
+	// Register the group on ALL daemons
+	group, err := cluster.DaemonAt(0).NewGroup("test-remove-empty", 3000000, getter)
+	require.NoError(t, err)
+	_, err = cluster.DaemonAt(1).NewGroup("test-remove-empty", 3000000, getter)
+	require.NoError(t, err)
+
+	// Test RemoveKeys with no keys - should not error
+	err = group.RemoveKeys(ctx)
+	require.NoError(t, err)
+}
+
+func TestRemoveKeysWithSlice(t *testing.T) {
+	ctx := context.Background()
+
+	err := cluster.Start(ctx, 2, groupcache.Options{})
+	require.NoError(t, err)
+	defer func() { _ = cluster.Shutdown(ctx) }()
+
+	getter := groupcache.GetterFunc(func(ctx context.Context, key string, dest transport.Sink) error {
+		return dest.SetString(fmt.Sprintf("value-%s", key), time.Now().Add(time.Minute*5))
+	})
+
+	// Register the group on ALL daemons
+	group, err := cluster.DaemonAt(0).NewGroup("test-remove-slice", 3000000, getter)
+	require.NoError(t, err)
+	_, err = cluster.DaemonAt(1).NewGroup("test-remove-slice", 3000000, getter)
+	require.NoError(t, err)
+
+	keys := []string{"key1", "key2", "key3"}
+
+	// Populate cache
+	for _, key := range keys {
+		var value string
+		err := group.Get(ctx, key, transport.StringSink(&value))
+		require.NoError(t, err)
+	}
+
+	// Test RemoveKeys with slice expansion
+	err = group.RemoveKeys(ctx, keys...)
+	require.NoError(t, err)
+}
+
+func TestRemoveKeysStats(t *testing.T) {
```

Comment:
We should either remove this test or add `GroupStats() GroupStats` to the `Group` interface and verify the stats. We can't make `group.Stats` private now unless we bump the major version, which I don't think we should do. However, adding `group.GroupStats()` method to the interface and implementation should accomplish what we want.